src/_pytest/_io/__init__.py

```diff
@@ -36,4 +36,6 @@ def _highlight(self, source):
         except ImportError:
             return source
         else:
-            return highlight(source, PythonLexer(), TerminalFormatter(bg="dark"))
+            if self.code_highlight:
```

Comment:
I didn't notice earlier, but it would be better to move this condition to above, i.e. `if not self.hasmarkup or not self.code_highlight`. This way, if `code_highlight` is false, the imports are not even attempted.